File: backend/src/main/java/com/legaltrack/service/DocumentService.java
package com.legaltrack.service;

import com.legaltrack.dto.document.CaseDocumentDto;
import com.legaltrack.enums.DocumentCategory;
import com.legaltrack.enums.DocumentVisibility;
import org.springframework.core.io.Resource;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface DocumentService {
    CaseDocumentDto uploadCaseDocument(Long caseId, MultipartFile file, DocumentCategory category, DocumentVisibility visibility, Long parentDocumentId);
    Resource downloadDocument(Long documentId);
    List<CaseDocumentDto> getDocumentsForCase(Long caseId);
    List<CaseDocumentDto> getDocumentVersionHistory(Long parentDocumentId);
    void deleteDocument(Long documentId);
}

File: frontend/src/services/documentService.js
import api from './api';

export const documentService = {
  uploadCaseDocument: async (caseId, file, category, visibility = 'LAWYER_AND_CLIENT', parentDocumentId = null) => {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('category', category);
    formData.append('visibility', visibility);
    if (parentDocumentId) formData.append('parentDocumentId', parentDocumentId);

    const res = await api.post(`/documents/cases/${caseId}/upload`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' }
    });
    return res.data;
  },

  getCaseDocuments: async (caseId) => {
    const res = await api.get(`/documents/cases/${caseId}`);
    return res.data;
  },

  downloadDocument: async (documentId, filename) => {
    const response = await api.get(`/documents/${documentId}/download`, {
      responseType: 'blob'
    });
    
    // Create download link
    const url = window.URL.createObjectURL(new Blob([response.data]));
    const link = document.createElement('a');
    link.href = url;
    link.setAttribute('download', filename || `document_${documentId}`);
    document.body.appendChild(link);
    link.click();
    link.remove();
  },

  getDocumentVersions: async (documentId) => {
    const res = await api.get(`/documents/${documentId}/versions`);
    return res.data;
  },

  deleteDocument: async (documentId) => {
    const res = await api.delete(`/documents/${documentId}`);
    return res.data;
  }
};
